pkg/storage/prunablesql/prunable_sql.go

```diff
@@ -0,0 +1,84 @@
+package prunablesql
+
+import (
+	copydir "github.com/otiai10/copy"
+	"gorm.io/gorm"
+
+	"github.com/iotaledger/hive.go/db"
+	"github.com/iotaledger/hive.go/ierrors"
+	"github.com/iotaledger/hive.go/log"
+	"github.com/iotaledger/hive.go/runtime/ioutils"
+	"github.com/iotaledger/hive.go/sql"
+	"github.com/iotaledger/iota-core/pkg/storage/utils"
+)
+
+type PrunableSQL struct {
+	errorHandler func(error)
+
+	databaseInstance *LockableGormDB
+	directory        string
+}
+
+func New(parentLogger log.Logger, directory string, errorHandler func(error)) *PrunableSQL {
+	baseDir := utils.NewDirectory(directory, true)
+
+	sqlDatabase, _, err := sql.New(
+		parentLogger.NewChildLogger("sql"),
+		sql.DatabaseParameters{
+			Engine:   db.EngineSQLite,
+			Path:     baseDir.Path(),
+			Filename: "tx_retainer.db",
```

Comment:
It seems a bit inconsistent to name the logger "sql" but the database file specific to the tx retainer. Maybe we can use `sqlite.db` as the name instead? I do think that it would be nice to have more specific names for the db files, but this seems like a hack. If we ever reuse the prunable sql for something else, it would be confusing.

Reply:
renamed it to `tx-retainer-db`